src/layer/lines.js

```diff
@@ -42,6 +42,15 @@ class Lines {
     get(index) {
         return this.cells[index];
     }
+
+    getCellAtRow(row) {
+        for (var i = 0; i < this.getLength(); i++){
```

Comment:
this could be a binary search right? or maybe something faster

Reply:
changed the implementation of the search

Reply:
implementation is changed to not use any search anymore